onlyswaps-verifier/src/eth.rs

```diff
@@ -0,0 +1,201 @@
+
+        transport.fetch_transfer_receipt(request_id).await
+    }
+
+    async fn fetch_transfer_params(
+        &self,
+        chain_id: u64,
+        request_id: FixedBytes<32>,
+    ) -> anyhow::Result<SwapRequestParameters> {
+        let transport = self
+            .networks
+            .get(&chain_id)
+            .ok_or(anyhow!("No chain transport for {}", chain_id))?;
+
+        transport.fetch_transfer_params(request_id).await
+    }
+
+    async fn submit_verification(
+        &self,
+        chain_id: u64,
+        verified_swap: VerifiedSwap,
+    ) -> anyhow::Result<()> {
+        let transport = self
+            .networks
+            .get(&chain_id)
+            .ok_or(anyhow!("No chain transport for {}", chain_id))?;
+
+        transport.submit_verified_swap(verified_swap).await
+    }
+}
+
+impl Network<DynProvider> {
+    pub async fn new(config: &NetworkConfig) -> anyhow::Result<Self> {
+        let url = config.rpc_url.clone();
+        let signer = PrivateKeySigner::from_slice(config.private_key.as_slice())?;
+        let own_addr = signer.address();
+        let provider = ProviderBuilder::new()
+            .with_gas_estimation()
+            .wallet(EthereumWallet::new(signer))
+            .connect_ws(WsConnect::new(url))
+            .await?
+            .erased();
+
+        println!("own addr: {}", own_addr);
+        Ok(Self {
+            chain_id: config.chain_id,
+            should_write: config.should_write,
+            router: RouterInstance::new(Address(config.router_address), provider.clone()),
+        })
+    }
+}
+impl<P: Provider> Network<P> {
+    pub async fn fetch_chain_state(&self) -> anyhow::Result<ChainState<RequestId>> {
+        let f = self.fetch_fulfilled_transfer_ids();
+        let v = self.fetch_verified_transfer_ids();
+        let (fulfilled, verified) = try_join(f, v).await?;
+        Ok(ChainState {
+            chain_id: self.chain_id,
+            fulfilled,
+            verified,
+        })
+    }
+    pub async fn fetch_transfer_params(
+        &self,
+        request_id: FixedBytes<32>,
+    ) -> anyhow::Result<SwapRequestParameters> {
+        Ok(self
+            .router
+            .getSwapRequestParameters(request_id)
+            .call()
+            .await?)
+    }
+
+    pub async fn fetch_transfer_receipt(
+        &self,
+        request_id: FixedBytes<32>,
+    ) -> anyhow::Result<TransferReceipt> {
+        let receipt = self.router.getSwapRequestReceipt(request_id).call().await?;
+        Ok(TransferReceipt {
+            chain_id: U256::from(self.chain_id),
+            request_id: receipt.requestId,
+            recipient: receipt.recipient,
+            src_chain_id: receipt.srcChainId,
+            token: receipt.token,
+            fulfilled: receipt.fulfilled,
+            solver: receipt.solver,
+            amount_out: receipt.amountOut,
+            fulfilled_at: receipt.fulfilledAt,
+        })
+    }
```

Comment:
This all seems very weird: we already have `omnievent` that should be providing all the data you need, no?

cc @azixus as for whether this should be here, or whether omnievent should be used instead.

Reply:
omnievent can listen for events, but it can't read contract state.
Not all the details required are in the events, and also events don't have the same guarantees for availability that contract state does.
In the medium term, we should find a way to listen to contract state changes with omnievent, but this will change things a lot